src/mpid/ch4/shm/xpmem/xpmem_pre.h

```diff
@@ -27,6 +28,12 @@ typedef struct MPIDI_XPMEM_seg {
     MPIR_cc_t refcount;         /* reference count of this seg */
 } MPIDI_XPMEM_seg_t;
 
+typedef struct MPIDI_XPMEM_cnt {
+    MPIR_OBJECT_HEADER;
+    OPA_int_t counter;
```

Comment:
Yes, we need it. Since `MPIR_Handle_direct_init` and `MPIR_Handle_indirect_init` use `MPIR_Handle_common` type to iterate objects in the pool, and `MPIR_Handle_common` is 16 bytes,  the minimal size of obj in the obj pool should be 16 bytes.